### PR TITLE
Use helper for mandatory relation pairs

### PR DESCRIPTION
**Pull request by sed-i**
## Issue
Mandatory relation pairs were modeled incorrectly.

## Solution
Use helper for mandatory relation pairs.

In tandem with:
- https://github.com/canonical/cos-lib/pull/25
- https://github.com/canonical/grafana-agent-k8s-operator/pull/286

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
-# 
+#
 # FIXME: Packing the charm with 2.2.0+139.gd011d92 will not include dependencies in PYDEPS key:
 # https://chat.charmhub.io/charmhub/pl/wngp665ycjnb78ar9ojrfhxjkr
 # That's why we are including cosl here until the bug in charmcraft is solved
-cosl
+#cosl
+git+https://github.com/canonical/cos-lib@feature/mandatory-relation-pairs
```
Comment (Abuelodelanada) [inline requirements.txt:8]: Why do not we wait for cosl to be released to pypi so we do not have to create another PR ?

File: src/charm.py [hunk 2]
```diff
@@ -154,11 +154,14 @@ class GrafanaAgentMachineCharm(GrafanaAgentCharm):
     mandatory_relation_pairs = {
         "cos-agent": [  # must be paired with:
             {"grafana-cloud-config"},  # or
-            {"send-remote-write", "logging-consumer", "grafana-dashboards-provider"},
+            {"send-remote-write"},  # or
+            {"logging-consumer"},  # or
+            {"grafana-dashboards-provider"},
```
Comment (Abuelodelanada) [inline src/charm.py:159]: Are we sure this refactor is ok?

`cos-agent` incoming relation needs an outgoing `grafana-cloud-config` or these three: `send-remote-write`, `logging-consumer`, `grafana-dashboards-provider` in order to set the `ActiveStatus`

File: src/charm.py [hunk 3]
```diff
@@ -154,11 +154,14 @@ class GrafanaAgentMachineCharm(GrafanaAgentCharm):
...
         ],
         "juju-info": [  # must be paired with:
             {"grafana-cloud-config"},  # or
-            {"send-remote-write", "logging-consumer"},
+            {"send-remote-write"},  # or
+            {"logging-consumer"},
```
Comment (Abuelodelanada) [inline src/charm.py:164]: Are we sure this refactor is ok?

`juju-info` incoming relation needs an outgoing `grafana-cloud-config` or these two: `send-remote-write`, `logging-consumer` in order to set the `ActiveStatus`

File: requirements.txt (hunk 1 above)
Comment (sed-i) [reply requirements.txt]: Yep, done!

File: src/charm.py (hunk 2 above)
Comment (sed-i) [reply src/charm.py:159]: I was surprised too but turns out it's desired behavior: https://github.com/canonical/grafana-agent-k8s-operator/pull/205

File: src/charm.py (hunk 3 above)
Comment (sed-i) [reply src/charm.py:164]: I was surprised too but turns out it's desired behavior: https://github.com/canonical/grafana-agent-k8s-operator/pull/205

Comment (Abuelodelanada) [reply src/charm.py:164]: Oh!! I did bot remember that!!! 🤦🏼‍♂️
